src/utils/tokens.ts

```diff
@@ -271,15 +271,15 @@ const tokens = {
     mintAmount: OutputValueType,
     depositPercent: number = TOKEN_DEPOSIT_PERCENTAGE
   ): OutputValueType {
-    return Math.ceil(depositPercent * mintAmount);
+    return BigInt(Math.ceil(depositPercent * Number(mintAmount)));
```

Comment:
Won't this `Number(mintAmount)` affect the deposit amount if the `mintAmount` is too big?

Reply:
Yes it will. This is exactly the same problem as described in https://github.com/HathorNetwork/hathor-wallet-lib/pull/708#discussion_r1857469627, and again it conforms to the protocol implementation in Python, which does this conversion implicitly, losing precision:

https://github.com/HathorNetwork/hathor-core/blob/4da0fb9186793ee1116eab39915441909fcf1c28/hathor/transaction/util.py#L54-L59

In practice, this can never be a problem in mainnet because the maximum number of existing tokens does not reach the boundary where a double loses integer precision. For Ekvilibro, I think it's possible that there exists a number of tokens that would lose precision here, but again it's not too big of a problem because this is compatible with what the network (hathor-core) expects.

The worse that could happen is that they could deposit a bit more tokens than they withdraw after a mint and melt. This is already the expected behavior for low amounts which would require deposits of less than 1 HTR (you deposit 1 and withdraw 0). Even then it's not an issue because we created an amount of native tokens that more than covers the amount of custom tokens they expect to mint. To be safe, we should keep this is mind for other use cases, though.

Reply:
Makes sense, I ran the test with python and js

```python
>>> math.ceil(abs(2**63 * 0.01))
92233720368547760
```

```js
> Math.ceil(Number(2n**63n) * 0.01)
92233720368547760
```

Can you just add a comment regarding this in the code, please?

Reply:
Sure! Done in https://github.com/HathorNetwork/hathor-wallet-lib/pull/708/commits/3f0df644ff4b75edac3dd0e800d5963cb25f1f5f